pkgs/applications/editors/emacs/elisp-packages/manual-packages/lsp-bridge/default.nix

```diff
@@ -0,0 +1,124 @@
+{ lib
+, python3
+, trivialBuild
+, fetchFromGitHub
+, substituteAll
+, markdown-mode
+, posframe
+, yasnippet
+, git
+, go
+, gopls
+, pyright
+, tempel
+, runCommand
+, lsp-bridge
+, unstableGitUpdater
+}:
+
+let
+  python = python3.withPackages (ps: with ps; [
+    epc
+    orjson
+    sexpdata
+    six
+  ]);
+in
+trivialBuild {
+  pname = "lsp-bridge";
+  version = "unstable-2023-03-06";
+
+  src = fetchFromGitHub {
+    owner = "manateelazycat";
+    repo = "lsp-bridge";
+    rev = "b8dcb59790f390767fb5ceb2f8b11c6938456738";
+    sha256 = "sha256-6KyvRwwIMVolc8ufl0sqMR35d2MLLDpeDf86690LP+U=";
+  };
+
+  patches = [
+    # Hardcode the python dependencies needed for lsp-bridge, so users
+    # don't have to modify their global environment
+    (substituteAll {
+      src = ./hardcode-python-dependencies.patch;
+      inherit python;
+    })
+  ];
+
+  packageRequires = [
+    markdown-mode
+    posframe
+    yasnippet
+  ];
+
+  buildInputs = [ python ];
+
+  # Check phase includes byte compilation
+  # https://github.com/manateelazycat/lsp-bridge/pull/557/commits/264869f55f595db45602c707f766a090d137c9cb
+  buildPhase = null;
+
+  checkInputs = [
+    git
+    go
+    gopls
+    pyright
+    tempel
+  ];
+
+  doCheck = true;
+  checkPhase = ''
+    runHook preCheck
+
+    mkfifo test.log
+    cat < test.log &
+    HOME=$(mktemp -d) python -m test.test
+
+    runHook postCheck
+  '';
+
+  installPhase = ''
```

Comment:
Is it possible to support `;;;###autoload` as what `elpaBuild` [does][1]?

[1]: https://github.com/NixOS/nixpkgs/blob/acc86a93168e272538f4ce459eaef3f58848ebd0/pkgs/build-support/emacs/elpa.nix#L26

Reply:
Oh, yep I can add support for this! I'm surprised that `trivialBuild` doesn't already do this? For now I'll just add it for lsp-bridge, but I think we should do this automatically in `trivialBuild`.

Reply:
After looking into this more, I don't really see a good way to do this on just lsp-bridge without special hacks. We could install to lsp-bridge to the elpa directory so Emacs knows to automatically pickup the autoloads - but this isn't an elpa package so that doesn't really make any sense. Or we could update the emacs wrapper to generate autoloads for all packages (not just melpa ones) - but in that case we might as well do it for all packages and not just lsp-bridge.

Reply:
I think I'd just want to leave adding support for this in another PR, because I'm not sure of the impact of the general solution, and it's an issue for all manual packages right now, not just lsp-bridge.

Reply:
I suggest
1. using `elpaBuild` instead of `trivialBuild`,
2. using `postInstall` (or `preInstall`) instead of `installPhase`

Reply:
Oh thanks! I already knew what ELPA stands for, but my misunderstanding was just thinking ELPA was more structured like other package formats. I didn't realize it could simply be an archive of the lisp file(s) without required metadata like `<PACKAGE NAME>-pkg.el`.

I still don't think using `elpaBuild` is the right approach though. I still want to hard-code the python dependencies and run tests, but `elpaBuild` doesn't support that because all it does is run `package-unpack` on the `$src` tarball.

It looks like maybe we could generate a tarball and hand it off to `elpa2nix.nix` like `melpaBuild` does though?

Reply:
Thinking through this a bit more, @adisbladis would make sense to generate ELPA packages for all manual packages - because it seems like most of them use the `trivialBuild` right now.

By not making them ELPAs we loose out on things like support for things like [magic autoload comments](https://www.gnu.org/software/emacs/manual/html_node/elisp/Autoload.html#Autoload).

Reply:
@jian-lin I'm still not sure if this problem should block merging this PR because it's wider issue, and I think it'd be better to work towards a general solution and not just something specific to lsp-bridge.

Reply:
> I didn't realize it could simply be an archive of the lisp file(s) without required metadata like <PACKAGE NAME>-pkg.el (and that package-unpack automatically generates the autoloads file)

A [multi-file package][mfp] has some requirements and one of them is a `NAME-pkg.el` file. `NAME-pkg.el` can only be auto-generated for a [simple package][sp].

> I still don't think using elpaBuild is the right approach though. I want to hard-code the python dependencies and run tests, but elpaBuild doesn't support that because all it does is run package-unpack on the $src tarball and doesn't actually run the unpackPhase.

Oh, I see, `elpaBuild` is not very suitable for this package.

Then I suggest
1. making a tarball which satisfies the requirements of a [multiple file package][mfp],
2. manually running this code
```bash
emacs --batch -Q -l ${./elpa2nix.el} \
  -f elpa2nix-install-package \
  "$path_to_tarball" "$out/share/emacs/site-lisp/elpa"
```

> It looks like maybe we could generate a tarball and hand it off to elpa2nix.nix like melpaBuild does though?

Yeah.

> I'm still not sure if this problem should block merging this PR because it's a wider issue

I don't think it's a blocker neither. It is good to have and can be done without much effort.

[mfp]: https://www.gnu.org/software/emacs/manual/html_node/elisp/Multi_002dfile-Packages.html
[sp]: https://www.gnu.org/software/emacs/manual/html_node/elisp/Simple-Packages.html